dvc/api.py

```diff
@@ -45,29 +46,31 @@ def __getattr__(self, name):
         )
 
 
-def _open(path, repo=None, remote=None, mode="r", encoding=None):
-    with _make_repo(repo) as _repo:
+def _open(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
+    with _make_repo(repo, rev=rev) as _repo:
         abspath = os.path.join(_repo.root_dir, path)
         with _repo.open(
             abspath, remote=remote, mode=mode, encoding=encoding
         ) as fd:
             yield fd
 
 
-def read(path, repo=None, remote=None, mode="r", encoding=None):
+def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
```

Comment:
btw, what will happen if I call it for a directory?

Reply:
It should work as long as `Repo(some_dir, rev=rev)` works, which is problematic, but that is a separate issue - #2201.

Reply:
An example where that won't work - remote configuration differs in a branch. Now a config file is always read from a actual fs, not git tree.